apps/web/vibes/soul/sections/product-detail/product-detail-form.tsx

```diff
@@ -175,10 +177,13 @@ function FormField({
 
   const handleChange = useCallback(
     (value: string) => {
-      void setParam(value);
+      if (field.deepLink === true) {
+        void setParam(value);
```

Comment:
Instead of conditionally calling `setParam` we probably want to completely exclude the parser from `nuqs`

Reply:
This is not performant, instead of doing nothing, this actually triggers a rerender of the form as if `params` would've changed. 

I will stick with the conditional logic.

Reply:
Changed to use `useQueryStates` which works as expected.